src/checkers/inference/InferenceValidator.java

```diff
@@ -1,402 +1,24 @@
 package checkers.inference;
 
-/*>>>
-import org.checkerframework.checker.compilermsgs.qual.CompilerMessageKey;
-*/
 
 import org.checkerframework.common.basetype.BaseTypeChecker;
-import org.checkerframework.common.basetype.TypeValidator;
-import org.checkerframework.framework.source.Result;
+import org.checkerframework.common.basetype.BaseTypeValidator;
 import org.checkerframework.framework.type.AnnotatedTypeFactory;
-import org.checkerframework.framework.type.AnnotatedTypeMirror;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedArrayType;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedDeclaredType;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedPrimitiveType;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedTypeVariable;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedWildcardType;
-import org.checkerframework.framework.type.AnnotatedTypeParameterBounds;
-import org.checkerframework.framework.type.visitor.AnnotatedTypeScanner;
-import org.checkerframework.javacutil.Pair;
-import org.checkerframework.javacutil.TreeUtils;
-
-import java.util.List;
-
-import javax.lang.model.element.TypeElement;
-import javax.lang.model.type.TypeKind;
-
-import com.sun.source.tree.AnnotatedTypeTree;
-import com.sun.source.tree.ExpressionTree;
-import com.sun.source.tree.IdentifierTree;
-import com.sun.source.tree.NewClassTree;
-import com.sun.source.tree.ParameterizedTypeTree;
-import com.sun.source.tree.Tree;
-import com.sun.source.tree.VariableTree;
 
 /**
  * A visitor to validate the types in a tree.
  */
-public class InferenceValidator extends AnnotatedTypeScanner<Void, Tree> implements TypeValidator {
-    protected boolean isValid = true;
+public class InferenceValidator extends BaseTypeValidator {
 
-    protected final BaseTypeChecker checker;
-    protected final InferenceVisitor<?, ?> visitor;
-    protected final AnnotatedTypeFactory atypeFactory;
+    protected boolean infer;
```

Comment:
Update javadoc. Explain the purpose of this field.

Think about how to share this flag.